q2_emperor/_plot.py

```diff
@@ -76,12 +79,20 @@ def plot(output_dir: str, pcoa: skbio.OrdinationResults,
 
 def procrustes_plot(output_dir: str, reference_pcoa: skbio.OrdinationResults,
                     other_pcoa: skbio.OrdinationResults,
-                    metadata: qiime2.Metadata, custom_axes: str = None,
+                    metadata: qiime2.Metadata, m2_stats: pd.DataFrame = None,
+                    custom_axes: str = None,
                     ignore_missing_samples: bool = False) -> None:
+    info = None
+    if m2_stats is not None:
+        m2 = round(m2_stats['true M^2 value'][0], 5)
+        trials = m2_stats['number of Monte Carlo trials'][0]
+        dec_places = math.ceil(math.log10(trials))
+        p_val = round(m2_stats['p-value for true M^2 value'][0], dec_places)
+        info = 'M&sup2; = ' + str(m2) + ' p-value = ' + str(p_val)
```

Comment:
Hey @ahdilmore (@ElDeveloper), I don't think this code is working quite as expected (unless of course, I misunderstood how it should be working. @lizgehret & I were testing this new viz functionality out (its super cool, btw!) and noted that the rounding is a bit unpredictable, and I think this makes sense because of how floating point numbers work. Below is a suggestion that uses string formatting to get to what I think are the desired results, please take a peek and let me know what you think.

```suggestion
        m2 = '%.5f' % m2_stats['true M^2 value'][0]
        trials = m2_stats['number of Monte Carlo trials'][0]
        dec_places = math.ceil(math.log10(trials))
        # Because the number of p-val dec places is dynamic, we need to dynamically
        # build the string template up.
        p_val_tmpl = '%%.df' % dec_places
        p_val = p_val_tmpl % m2_stats['p-value for true M^2 value'][0]
        info = 'M&sup2; = %s p-value = %s' % (m2, p_val)
```

PS - @lizgehret will send a screenshot from our tests, too follow.

Reply:
@ahdilmore here's what we are currently seeing in the viz:
![p-val-decimals](https://user-images.githubusercontent.com/54517601/114931292-e849f080-9dea-11eb-8095-fe5c34584bdc.png)

Reply:
Thanks for sharing that, @lizgehret! So, I think the problem in that screenshot is the p-val - that _should_ have 3 zeros, after the decimal point, if I'm reading this code right - `0.000`, because we generated the test data with 1000 trials.